src/WinRT.Runtime/Projections/Bindable.net5.cs

```diff
@@ -396,22 +398,84 @@ internal unsafe interface IEnumerable : global::System.Collections.IEnumerable,
 #pragma warning disable CA2257 // This member is a type (so it cannot be invoked)
         public sealed class AdaptiveFromAbiHelper : FromAbiHelper, global::System.Collections.IEnumerable
 #pragma warning restore CA2257
-        {
-            private readonly Func<IWinRTObject, global::System.Collections.IEnumerator> _enumerator;
-
+        {
+#if NET8_0_OR_GREATER
+            private readonly MethodInvoker _enumerator;
+#else
+            private readonly MethodInfo _enumerator;
+#endif
+
+#if NET
+            [DynamicDependency(DynamicallyAccessedMemberTypes.PublicMethods, typeof(IEnumerable<>))]

```

Comment:
@MichalStrehovsky I could use a second pair of eyes to double check this is actually safe. I'm not 100% sure 😅